openvdb_houdini/SOP_OpenVDB_Merge.cc

```diff
@@ -0,0 +1,296 @@
+            "point"     "Nearest"
+            "linear"    "Linear"
+            "quadratic" "Quadratic"
+        }
+    }
+    groupsimple {
+        name    "mergemodegroup"
+        label   "Merge Mode"
+
+        parm {
+            name    "fogmode"
+            label   "Fog Mode"
+            type    ordinal
+            default { "add" }
+            menu {
+                "none"          "None"
+                "add"           "Add"
+                "multiply"      "Multiply"
+                "maximum"       "Maximum"
+                "minimum"       "Minimum"
+                "topounion"     "Activity Union"
+                "topointersect" "Activity Intersection"
+            }
+            joinnext
+        }
+        parm {
+            name    "scalarmode"
+            label   "Scalar Mode"
+            type    ordinal
+            default { "add" }
+            menu {
+                "none"          "None"
+                "add"           "Add"
+                "multiply"      "Multiply"
+                "maximum"       "Maximum"
+                "minimum"       "Minimum"
+                "topounion"     "Activity Union"
+                "topointersect" "Activity Intersection"
+            }
+        }
+        parm {
+            name    "distancemode"
+            label   "Distance Mode"
+            type    ordinal
+            default { "sdfunion" }
+            menu {
+                "none"          "None"
+                "sdfunion"      "SDF Union"
+                "sdfintersect"  "SDF Intersect"
+                "topounion"     "Activity Union"
+                "topointersect" "Activity Intersection"
+            }
+            joinnext
+        }
+        parm {
+            name    "vectormode"
+            label   "Vector Mode"
+            type    ordinal
+            default { "add" }
+            menu {
+                "none"          "None"
+                "add"           "Add"
+                "multiply"      "Multiply"
+                "topounion"     "Activity Union"
+                "topointersect" "Activity Intersection"
+            }
+        }
+        parm {
+            name    "boolmode"
+            label   "Bool Mode"
+            type    ordinal
+            default { "topounion" }
+            menu {
+                "none"          "None"
+                "topounion"     "Activity Union"
+                "topointersect" "Activity Intersection"
+            }
+            joinnext
+        }
+        parm {
+            name    "pointsmode"
+            label   "Points Mode"
+            type    ordinal
+            default { "pointmerge" }
+            menu {
+                "none"          "None"
+                "pointmerge"    "Merge"
+                "topounion"     "Activity Union"
+                "topointersect" "Activity Intersection"
+            }
+        }
+        parm {
+            name        "useallgridsfortopo"
+            label       "Use all Grids for Activity Modes"
```

Comment:
Should be
Use All VDBs for Activity Operations
Note that All is capitalized in Title Case.  Grids -> VDBs and Modes -> Operations.
I have no idea what this does, though, so I think it needs better text :>

Reply:
The "collation" modes groups the grids you can merge (type and class, etc). That's mainly because there's no obvious or useful way to value merge a vector grid with a level set grid for instance. However, for topology modes, it's valid to be able to "activity union" a set of grids of all different types and classes, so that feels like a natural extension to want to include.

The alternative workflow here is to create a temporary bool grid, use the VDB Activate SOP to activity union all input grids, then use another VDB Activate SOP to activity union all (or specific) input grids against this temporary grid, which is somewhat convoluted.

The more I think about this, the more I think perhaps the most intuitive way of offering this could be to add additional drop-down options:

* Activity Union
* Activity Intersect
* Activity Union (All Grids)
* Activity Intersect (All Grids)